blues/simulation.py

```diff
@@ -380,10 +382,24 @@ def freeze_radius(cls, structure, system, freeze_distance=5.0*unit.angstrom,
         #Select the LIG and atoms within 5 angstroms, except for WAT or IONS (i.e. selects the binding site)
         if hasattr(freeze_distance, '_value'): freeze_distance = freeze_distance._value
         selection = "(%s<:%f)&!(%s)" % (freeze_center,freeze_distance,freeze_solvent)
+        logger.info('Inverting parmed selection for freezing: %s' % selection)
         site_idx = cls._amber_selection_to_atom_indices_(structure, selection)
         #Invert that selection to freeze everything but the binding site.
         freeze_idx = set(range(system.getNumParticles())) - set(site_idx)
 
+        #Check if freeze selection has selected all atoms
+        if len(freeze_idx) == system.getNumParticles():
+            err = 'All %i atoms appear to be selected for freezing. Check your atom selection.' % len(freeze_idx)
+            logger.error(err)
+            sys.exit(1)
+
+        #Ensure that the freeze selection is larger than the center selection point
+        center_idx = cls._amber_selection_to_atom_indices_(structure, freeze_center)
+        if len(site_idx) <= len(center_idx):
+            err = "%i unfrozen atoms is less than (or equal to) the number of atoms from the selection center '%s' (%i atoms). Check your atom selection." %(len(site_idx), freeze_center, len(center_idx))
```

Comment:
I don't quite understand this check/the associated messages. What does it mean "larger than the center selection point"? A point is, er, a set of coordinates, right? Maybe you mean "less than or equal to the number of atoms provided to serve as the center of the selection"? (This applies to both the comment and the logger message).

I also think that, above, the check whether the freeze selection has selected ALL atoms is probably not careful enough; I would guess that it's safe to say that you should issue a warning if you select, say, more than 75% (??) of the atoms in the system and throw an exception if you select more than 99% (??). Exact thresholds subject to debate (you might want to check what this is for your system), but it's probably rare you'd select ALL the atoms even if you mess up, whereas a mess up will often result in selecting too many atoms.